temoa/_internal/table_writer.py

```
@@ -200,321 +228,291 @@ def clear_scenario(self) -> None:
+
+        self._bulk_insert('output_emission', records)
         self.con.commit()
 
+    def write_capacity_tables(self, model: TemoaModel, iteration: int | None = None) -> None:
+        cap_data = poll_capacity_results(model=model)
+        self._insert_capacity_results(cap_data=cap_data, iteration=iteration)
+
     def _insert_capacity_results(self, cap_data: CapData, iteration: int | None) -> None:
-        if not self.tech_sectors:
+        if self.tech_sectors is None:
             raise RuntimeError('tech sectors not available... code error')
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)